### ISSUE TITLE
Nick highlight should match word boundaries

### ISSUE THREAD

**Issue by calzoneman** [calzoneman/sync#819]
Been reported a few times, should be fixed at some point.  The current highlight check is `indexOf()` but it should do a regular expression match of word boundaries to avoid triggering on usernames that are substrings of other words.

**Comment by twilight-sparkle-irl** [comment]
It also, for the record, highlights when the username is in HTML and not the actual text of the message.

**Comment by calzoneman** [comment]
Yeah, emotes trigger it as well -- maybe walking the DOM tree (after the message is parsed) for TextNodes might address that.

**Comment by twilight-sparkle-irl** [comment]
Why not just use `innerText`/`textContent`?

**Comment by calzoneman** [comment]
Inspecting `innerText` on the root of the message's DOM may have subtle edge cases around word boundaries, e.g. `<span>foo</span><span>bar</span>` getting merged into `foobar` when in the source HTML it was semantically 2 separate fragments of text.

I also realized if I strip highlights from HTML attributes after emote conversion it might break some legitimate use cases -- for example, people who assign emotes to people's usernames to display an icon instead of text (but still intend to highlight).

**Comment by twilight-sparkle-irl** [comment]
Perhaps sending the initial unfiltered chat message and doing mentions based on that would work? Most of the cases where people want to mention someone would involve them typing it in themselves, so I figure that'd be a better solution.
